### DISCUSSION TITLE
Allow setting a started date

### DISCUSSION THREAD

**Discussion by davidskeck** [iSoron/uhabits#1261] [Feature Requests]
For all my habits I have a start date several years in the past. It would be great if I could set that somehow in the app and not have to click a ton of dates to mark them as complete. Currently all the start dates are wrong. (I haven't missed a day on any of the tasks)

**Comment by hiqua** [comment]
You mean that you started tracking your habits in the app after you formed the habits?

**Comment by davidskeck** [comment]
Yes, that's correct.

**Comment by ke352802081770314** [comment]
I feel that entering a start date to express that I had a perfect history since then can be misunderstood and might be entered by mistake and it would be surprising that it makes me perfect. Actually I am a bit puzzled why you need to track it when you are perfect at that habit already, is it because you want to prevent yourself from slipping?
it could help people new to the app to have a mass-add feature for past achievements, like "tick all days from day X to day Y" to save some work. This could be next to the "edit" button on the calendar of the individual habit view.

**Comment by davidskeck** [comment]
Yeah I'm trying to import my history from Streaks. But their format isn't supported. I mainly use uhabits to ensure that I complete each task daily and don't forget one. I would also like to know the length of the streak in days.

The solution you've described would work perfectly for me.

**Reply by hiqua** [reply]
It could be easier to support imports from the app you mentioned.

I think habits are mostly forward-looking, meaning it's nice to have stats, but the main goal is to stick with them in the future. So either:
* the user didn't use any habit app before (or were tracking them manually) and it doesn't really matter, it should be fine to forget about this data
* or they did, and in this case it'd be easier to support imports

@davidskeck do you have any export function in Streaks? Could you share a representative sample so that we can implement the import function? Preferably something with all the features there are, e.g. if there are numerical habits, skips etc., try to include them.

**Reply by davidskeck** [reply]
[streaks.csv](https://github.com/iSoron/uhabits/files/7865818/streaks.csv)
Sure, I've attached a small sample. This is as it existed when I last exported (2021, June 8th)

**Comment by lesderid** [comment]
I also wish this feature existed. I have been tracking my habits manually (in a text file) so an importer for a specific habit tracker wouldn't work for me, and I'm not really willing to tick hundreds of boxes manually. I think a good solution would be an option to add a certain date range (as suggested by @ke352802081770314).

Some kind of historical view is important for me, it's a strong motivator.

**Reply by davidskeck** [reply]
Exactly why I want the habits tracked. I find it motivating to see a long streak. :)
